sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriter.cs

```diff
@@ -81,6 +81,25 @@ public static BinaryData Write(object model, ModelReaderWriterOptions? options =
         }
     }
 
+    /// <summary>
+    /// Converts the value of a model into a <see cref="BinaryData"/>.
+    /// </summary>
+    /// <typeparam name="T">The type of the value to write.</typeparam>
+    /// <param name="model">The model to convert.</param>
+    /// <param name="options">The <see cref="ModelReaderWriterOptions"/> to use.</param>
+    /// <returns>A <see cref="BinaryData"/> representation of the model in the <see cref="ModelReaderWriterOptions.Format"/> specified by the <paramref name="options"/>.</returns>
+    /// <exception cref="FormatException">If the model does not support the requested <see cref="ModelReaderWriterOptions.Format"/>.</exception>
+    /// <exception cref="ArgumentNullException">If <paramref name="model"/> is null.</exception>
+    public static BinaryData Write<T>(T? model, ModelReaderWriterOptions? options = default)
```

Comment:
Can we add a nullability test here https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/System.ClientModel/tests/ModelReaderWriter/ModelReaderWriterTests.cs

Can we add test cases which exercise the various write paths as well for ModelAsStruct.

For the usual case we simply added a test class like this but your case is probably special since the return type 
 here is T so you won't maintain the T? typing during the flow.  https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/System.ClientModel/tests/ModelReaderWriter/Models/ModelAsStructTests.cs#L9.  You can probably just write a simpler version which reads from the json file and loads it into a ModelAsStruct? then call write and validate expected json.